redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt

```diff
@@ -549,6 +549,37 @@ abstract class AbstractFlexContainerTest<T : Any> {
     innerContainer2.modifier = Modifier.then(FlexImpl(1.0))
     verifySnapshot(outerContainer)
   }
+
+  @Test
+  fun testColumnWithChildModifierChanges() {
+    testContainerWithChildrenModifierChanges(FlexDirection.Column)
+  }
+
+  @Test
+  fun testRowWithChildModifierChanges() {
+    testContainerWithChildrenModifierChanges(FlexDirection.Row)
+  }
+
+  private fun testContainerWithChildrenModifierChanges(
+    flexDirection: FlexDirection,
+  ) {
+    assumeTrue(flexDirection in listOf(FlexDirection.Row, FlexDirection.Column))
+    val container = flexContainer(flexDirection)
+    container.width(Constraint.Fill)
+    container.height(Constraint.Fill)
+
+    val first = text(longText(), backgroundColor = Red)
+    first.modifier = MarginImpl(30.dp)
+
+    container.add(first)
+    container.add(text(mediumText(), backgroundColor = Green))
+    container.add(text(shortText(), backgroundColor = Blue))
+    container.onEndChanges()
+    verifySnapshot(container, "Margin")
+    first.modifier = Modifier
+    container.onEndChanges()
```

Comment:
I modified `TestFlexContainer` to expose the `children` property so that we can use `children.onModifierUpdated(0, widget)` inside [AbstractFlexContainerTest.kt](https://github.com/cashapp/redwood/pull/1995/files#diff-999b67f699c44ef77a269cd6e1dbe2b9537decb4d0060d8cca94d68c3e7c4c48).

Should we add `container.onEndChanges()` to `AbstractBoxTest`? This will require us to provide a `ChangeListener` implementation for the `Box` widget under test. In my opinion, this is a significant change. So, should we stick with using `children.onModifierUpdated` in both tests or work to support the `ChangeListener` for the `Box` widget?